### PR TITLE
Support various format of string value decoding/encoding

### PR DESCRIPTION
**Pull request by tboonIV**
- field key decoding/encoding
- blob decoding/encoding
- escape character in string decoding/encoding

### CODE REVIEW

File: cmdc/value.go [hunk 1]
```diff
@@ -154,6 +155,19 @@ func encodeStringValue(v string) ([]byte, error) {
...
+	if ch >= '0' && ch <= '9' {
+		return ch - '0'
+	}
+	if ch >= 'A' && ch <= 'F' {
+		return ch - 'A' + 10
+	}
+	if ch >= 'a' && ch <= 'f' {
+		return ch - 'a' + 10
+	}
+	panic("Invalid OctetString digit '" + string(ch) + "'. Valid digits are '0'-'9', 'A'-'F'")
```
Comment (lwlee2608) [inline cmdc/value.go:168]: return panic with error

File: cmdc/value.go [hunk 2]
```diff
@@ -169,13 +183,73 @@ func decodeStringValue(b []byte) (string, error) {
 			if err != nil {
 				panic("invalid string length")
 			}
-			str := string(b[idx+1 : idx+1+len])
-			return string(str), nil
+			var result []byte
+			for i := 1; i < len+1; i++ {
```
Comment (lwlee2608) [inline cmdc/value.go:187]: is it possible to put this blob of code into a new private function?

**Review by lwlee2608 — APPROVED** [review]

File: cmdc/value.go [hunk 3]
```diff
@@ -154,6 +155,48 @@ func encodeStringValue(v string) ([]byte, error) {
...
+}
+
+func decodeString(b []byte) (string, error) {
+	var result []byte
+	for i := 0; i < len(b); i++ {
+		c := b[i]
+		if c == '\\' {
+			// Escape Char '\'
+			i++
+			next1 := b[i]
```
Comment (lwlee2608) [inline cmdc/value.go:177]: out of bound access. example: code will panic with this input `"test\"`

File: cmdc/value.go [hunk 4]
```diff
@@ -154,6 +155,48 @@ func encodeStringValue(v string) ([]byte, error) {
...
+			if next1 == '\\' {
+				result = append(result, '\\')
+			} else {
+				i++
+				next2 := b[i]
+				b1, err := fromDigit(next1)
+				if err != nil {
+					return string(""), err
```
Comment (lwlee2608) [inline cmdc/value.go:185]: just do this:
```go
return "", err
```

File: cmdc/value.go [hunk 5]
```diff
@@ -154,6 +155,48 @@ func encodeStringValue(v string) ([]byte, error) {
...
+		c := b[i]
+		if c == '\\' {
+			// Escape Char '\'
+			i++
+			next1 := b[i]
+			if next1 == '\\' {
+				result = append(result, '\\')
+			} else {
+				i++
+				next2 := b[i]
```
Comment (lwlee2608) [inline cmdc/value.go:182]: out of bound access again

**Review by lwlee2608 — CHANGES_REQUESTED** [review]

**Comment by lwlee2608** [comment]
replace all 
```
    string("") 
```
to
```
    ""
```

**Comment by tboonIV** [comment]
> replace all
> 
> ```
>     string("") 
> ```
> 
> to
> 
> ```
>     ""
> ```

updated.

**Review by lwlee2608 — APPROVED** [review]

**Pull request merged**